common/asyncTask.py

```diff
@@ -0,0 +1,53 @@
+from typing import Callable, Coroutine, Tuple
+import asyncio
+
+
+class AsyncTask:
+    """
+    AsyncTasks are objects designed to encapsulate background tasks which run asynchronously.
+    This is very useful for performing asynchronous operations without creating race conditions and timing issues.
+    Note that at most only one instance of the background task will be active at any given time.
+    The background task must be provided to the initializer as a function that returns a coroutine function.
+
+    Example usage:
+        1) Send a reminder to the host every morning at 11 am to pick a movie.
```

Comment:
That is implemented, but not in a generic way. The files _upcomingMovie.py_ and _eventReminder.py_ implement the use-case you described respectively.

Adding the task to make these files more generic in the backlog.

Reply:
To add to the discussion: This is just a generic function to create asynchronous "threads", which can be used to achieve any time-related task. It might be good to make all existing use-cases mentioned in the previous comment more generic though.